sonar-ballerina-plugin/src/main/resources/io/ballerina/sonar/ballerina-5.html

```diff
@@ -0,0 +1,45 @@
+<p>
+    An class defined as isolated is similar to a module with isolated module-level variables. Mutable fields of an isolated object,
+</p>
+<ul>
+    <li>must be private and so can only be accessed using self</li>
+    <li>must only be accessed within a lock statement</li>
+    <li>lock statement must follow the same rules for self as for an isolated variable</li>
+    <li>must be initialized with an isolated expression</li>
+    <li>a field is mutable unless it is final and has a type that is a subtype of readonly</li>
+</ul>
+
+<p>
+    Marking a class as isolated helps to ensure the thread safety.
+</p>
+
+<h2>Noncompliant Code Example</h2>
+
+<pre>
+public class EvenNumber {
+    int i = 1;
+
+    public isolated function generate() returns int {
+        lock {
+            return self.i * 2;
+        }
+    }
+}
+</pre>
+
+<h2>Compliant Solution</h2>
+
+<p>
+    Mark public non-isolated classes as isolated if they are not the entry point of the program to ensure thread safety.
```

Comment:
```suggestion
    Mark public non-isolated classes as isolated if they are not the entry point of the program to ensure thread safety.
```

Classes cannot be entry points

Reply:
Fixed with f10b04b7337bff186e2de3ae2a65a2421c10884d